### ISSUE TITLE
Zygisk problem

### ISSUE THREAD

**Issue by nizonrox** [topjohnwu/Magisk#5499]
Device: Huawei P30 Pro VOG-L29
Android version: 9 (9.1.0.260(C431E4R2P2))
Magisk version name: Magisk, Canary
Magisk version code: 24102

I've updated to Canary branch since my last bug report with automatically closed.
My issue is that when I have zygisk on then the phone will crash at somepoint on and I would have to boot into recovery for root again.
After updating to Canary I've still had crashes but not of the same degree as it just reboots into root.
Crashing would happen when i clicked the clock widget or initiate a call or opening an app.

Before Canary I had "/system/bin/linker64" in all my tombstones, after getting Canary my phone hasnt made a new tombstone.
Below i have attatched my most recent tombstone and my most recent magisk.log.bak If you need more then lmk but there are no custom roms for P30 pro.
[P30Pro_LogTomb.zip](https://github.com/topjohnwu/Magisk/files/8154562/P30Pro_LogTomb.zip)
[Auto-Closed Issue v24.1](https://github.com/topjohnwu/Magisk/issues/5494)

**Comment by nizonrox** [comment]
[magisk24103.zip](https://github.com/topjohnwu/Magisk/files/8163472/magisk24103.zip)
Same problem on 24103; A guy on XDA forums has the same problem on a Huawei P10 (Not sure if lite/pro).

**Comment by nizonrox** [comment]
> the tomestone is not about the reboot.

What about the magisk logs? not useful?

**Comment by yujincheng08** [comment]
I don't think magisk log of this boot will show the reason of the last reboot?

**Comment by nizonrox** [comment]
> I don't think magisk log of this boot will show the reason for the last reboot?
[logcat.txt](https://github.com/topjohnwu/Magisk/files/8170075/logcat.txt)

I finally got a logcat of it happening, but Please let me know what to try to supply
Nizonrox#7777 on discord :)

**Comment by yujincheng08** [comment]
This log is not about the reboot either.

**Comment by nizonrox** [comment]
After i changed to Canary, I don't get tombstones anymore and it reboots straight into rooted. I'm trying to look for console-ramoops but its not in its default location....

**Comment by Lenny5Aces** [comment]
> [magisk24103.zip](https://github.com/topjohnwu/Magisk/files/8163472/magisk24103.zip) Same problem on 24103; A guy on XDA forums has the same problem on a Huawei P10 (Not sure if lite/pro).

Yes, that guy is me. I got a "normale" P10 (not LIght nor Pro - rebranded L09 -> L29) with Android 9.1.0.297 (C432E5R1P9), the latest Magisk Alpha 24201 (should I switch to Canary for debugging? I'm afraid I still don't know the differences between these two).

When I activate Zygisk in settings  my device crashes randomly when starting any app (as soon as I click the app the screen gets dark).

I saved a Magisk log file but I'm not sure if anyone can read something from it:
[magisk_log_2022-03-02T19:23:51.log](https://github.com/topjohnwu/Magisk/files/8175813/magisk_log_2022-03-02T19.23.51.log)

**Comment by nizonrox** [comment]
[console-ramoops-0.txt](https://github.com/topjohnwu/Magisk/files/8191240/console-ramoops-0.txt)

I just updated to 24201 and got a console-ramoops after a crash. Please take a look! :)

**Comment by yujincheng08** [comment]
crash in this fuction
https://github.com/TadiT7/android_kernel_huawei_TNY/blob/e49326f033627394a73ecd32bca08b0d1b5821d6/kernel/cgroup_workingset.c#L3358

**Comment by yujincheng08** [comment]
i think its a kernel bug that accidentally triggered by zygisk

**Comment by yujincheng08** [comment]
seems like you have the kernel source so maybe you can add some `printk` to debug which line of code causes the kernel panic and find the fix correspondingly.

**Comment by nizonrox** [comment]
I have gone with updating and hope they fixed it at somepoint, I'm on 11.0 now but magisk now says i have a ramdisk? before i was patching my recovery_ramdisk. what do i do now?

**Comment by Kevin2765742** [comment]
the same to u

**Comment by yangmx** [comment]
close zram
1. create file: /data/adb/service.d/1.sh
2. fill it
#!/system/bin/sh
swapoff /dev/block/zram0
sysctl -w vm.swappiness=0
3. chmod 777 /data/adb/service.d/1.sh
4. execute it, or reboot

**Comment by maxwag9** [comment]
How the hell did you even root a Huawei? I have the Huawei p 30 Lite.
